python_modules/dagster/dagster/_core/definitions/schedule_definition.py

```diff
@@ -564,20 +586,26 @@ def __init__(
                 "Dagster recognizes standard cron expressions consisting of 5 fields."
             )
 
+        self._target: Optional[Union[RepoRelativeTarget, DirectTarget]] = None
         if job is not None:
-            self._target: Union[DirectTarget, RepoRelativeTarget] = DirectTarget(job)
-        else:
+            self._target = DirectTarget(job)
+        elif job_name:
             self._target = RepoRelativeTarget(
                 job_name=check.str_param(job_name, "job_name"),
                 op_selection=None,
             )
+        elif asset_selection:
+            self._target = None
 
         if name:
             self._name = check_valid_name(name)
         elif job_name:
             self._name = job_name + "_schedule"
         elif job:
             self._name = job.name + "_schedule"
+        else:
+            # TODO: better default name for asset selection or should we always ask the users to specify a name when asset_selection is provided?
+            self._name = "asset_selection_schedule"
```

Comment:
this wasn't a problem in sensor because sensor always has a decorated function. in schedule cases, a user can create a schedule via `basic_schedule = ScheduleDefinition(asset_selection=AssetSelection.all(), cron_schedule="0 0 * * *")`. in that case, we would construct the schedule name based on job name to be `${job_name}_schedule`. but when there's no job provided, i couldn't think of a great name to fill in. a few ideas:
- `assets_${first_5_or_less_char_of_asset_selection}_schedule`
- `${first_10_or_less_char_of_asset_selection}_schedule`

curious to see if folks have ever thought of some kind of "asset selection name/identifier" string value? cc @sryza